core/services/relay/evm/ocr3_capability_provider.go

```diff
@@ -14,3 +26,163 @@ type ocr3CapabilityProvider struct {
 func (o *ocr3CapabilityProvider) OCR3ContractTransmitter() ocr3types.ContractTransmitter[[]byte] {
 	return o.transmitter
 }
+
+var _ LogDecoder = &ocr3CapabilityLogDecoder{}
+
+type ocr3CapabilityLogDecoder struct {
+	eventName string
+	eventSig  common.Hash
+	abi       *abi.ABI
+}
+
+// Modified newOCR2AggregatorLogDecoder to use OCR3Capability ABI
+func newOCR3CapabilityLogDecoder() (*ocr3CapabilityLogDecoder, error) {
+	const eventName = "ConfigSet"
+	abi, err := ocr3_capability.OCR3CapabilityMetaData.GetAbi()
+	if err != nil {
+		return nil, err
+	}
+	return &ocr3CapabilityLogDecoder{
+		eventName: eventName,
+		eventSig:  abi.Events[eventName].ID,
+		abi:       abi,
+	}, nil
+}
+
+func (d *ocr3CapabilityLogDecoder) Decode(rawLog []byte) (ocrtypes.ContractConfig, error) {
+	unpacked := new(ocr3_capability.OCR3CapabilityConfigSet)
+	err := d.abi.UnpackIntoInterface(unpacked, d.eventName, rawLog)
+	if err != nil {
+		return ocrtypes.ContractConfig{}, fmt.Errorf("failed to unpack log data: %w", err)
+	}
+
+	var transmitAccounts []ocrtypes.Account
+	for _, addr := range unpacked.Transmitters {
+		transmitAccounts = append(transmitAccounts, ocrtypes.Account(addr.Hex()))
+	}
+	var signers []ocrtypes.OnchainPublicKey
+	for _, addr := range unpacked.Signers {
+		addr := addr
+		signers = append(signers, addr[:])
+	}
+
+	return ocrtypes.ContractConfig{
+		ConfigDigest:          unpacked.ConfigDigest,
+		ConfigCount:           unpacked.ConfigCount,
+		Signers:               signers,
+		Transmitters:          transmitAccounts,
+		F:                     unpacked.F,
+		OnchainConfig:         unpacked.OnchainConfig,
+		OffchainConfigVersion: unpacked.OffchainConfigVersion,
+		OffchainConfig:        unpacked.OffchainConfig,
+	}, nil
+}
+
+func (d *ocr3CapabilityLogDecoder) EventSig() common.Hash {
+	return d.eventSig
+}
+
+var _ ocrtypes.OffchainConfigDigester = OCR3CapabilityOffchainConfigDigester{}
+
+// EVMOffchainConfigDigester forked to not assume signers are 20 byte addresses
+type OCR3CapabilityOffchainConfigDigester struct {
+	ChainID         uint64
+	ContractAddress common.Address
+}
+
+func (d OCR3CapabilityOffchainConfigDigester) ConfigDigest(cc ocrtypes.ContractConfig) (ocrtypes.ConfigDigest, error) {
+	signers := [][]byte{}
+	for _, signer := range cc.Signers {
+		signers = append(signers, signer)
+	}
+	transmitters := []common.Address{}
+	for i, transmitter := range cc.Transmitters {
+		if !strings.HasPrefix(string(transmitter), "0x") || len(transmitter) != 42 || !common.IsHexAddress(string(transmitter)) {
+			return ocrtypes.ConfigDigest{}, fmt.Errorf("%v-th evm transmitter should be a 42 character Ethereum address string, but got '%v'", i, transmitter)
+		}
+		a := common.HexToAddress(string(transmitter))
+		transmitters = append(transmitters, a)
+	}
+
+	return ocr3CapabilityConfigDigest(
```

Comment:
@archseer This shouldn't panic if we can avoid it, especially if the caller can return an error

Reply:
I copied that straight from libocr's `evmutils` and changed the signers type parameter, I assume it's panicking since it should really be a compile time check. If the digest size changed we can't recover from that